pkg/metrics/metrics.go

```diff
@@ -84,10 +87,23 @@ var ACMEClientRequestDurationSeconds = prometheus.NewSummaryVec(
 	[]string{"scheme", "host", "path", "method", "status"},
 )
 
+type registeredCertificates struct {
+	certificates map[*v1alpha1.Certificate]struct{}
+	mtx          sync.Mutex
+}
+
+// RegisteredCertificates holds the set of all certificates which are currently
+// registered by Prometheus
+var RegisteredCertificates = &registeredCertificates{
+	certificates: make(map[*v1alpha1.Certificate]struct{}),
+}
+
 type Metrics struct {
 	ctx context.Context
 	http.Server
 
+	registeredCertificates *registeredCertificates
```

Comment:
I'm confused here - why do we have a global `registeredCertificates` as well as a field in the Metrics struct?

Reply:
I was mimicking the other fields but I agree it is unnecessary actually